src/getLocale.ts

```diff
@@ -0,0 +1,45 @@
+import { isString } from './isString';
+import { storage } from './storage';
+import type { CountryCode } from './types/countries';
+
+const KEY = 'gu.geolocation';
+const URL = 'https://api.nextgen.guardianapps.co.uk/geolocation';
+
+// best guess that we have a valid code, without actually shipping the entire list
+const isValidCountryCode = (country: unknown) =>
+	isString(country) && /^[A-Z]{2}$/.test(country as string);
+
+const daysFromNow = (days: number) =>
+	new Date().getTime() + 60 * 60 * 24 * days;
+
+/**
+ * Fetches the user's current location as an ISO 3166-1 alpha-2 string e.g. 'GB', 'AU' etc
+ */
+
+export const getLocale = async (): Promise<CountryCode | null> => {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- it _is_ any
+	const stored = storage.local.get(KEY);
+
+	// if we've got a locale, return it
+	if (isValidCountryCode(stored)) return stored as CountryCode;
+
+	// use our API to get one
+	try {
```

Comment:
Why is this in a `try`?

Reply:
just in the event the fetch errors in some way. we dont really want to handle it since we can't do anything with it if it does, but this is intended to keep it quiet at least – do you think it's unecessary?

Reply:
I'm just curious really, and to check my understanding. I would expect `fetch` errors would need to be handled in a chained `catch` rather than a `catch` block?